### PR TITLE
Curtailment Solution

### LINKED ISSUE #212: Curtailment
Some sites get curtailed, and it would be good to have a way to store this. 

Typically site might get curtailed from a start and end date time, with a from and to time. 
There may be serval curtailments happening at the same time, for example 

Image

The new table could have
- from_date
- to_date
- from_time_utc
- to_time_utc
- curtialment_kw
- linked to site

We would need to 
- add read and write route to api, users would only have access to there site. We might want a edit/delete route too
- add method to read curtailment in forecast and clip forecast

### PR DESCRIPTION
**Pull request by Dakshbir**
# Pull Request

## Description

This PR implements curtailment functionality for PV sites, including:

1. New CurtailmentSQL model with fields:
   - from_date
   - to_date
   - from_time_utc
   - to_time_utc
   - curtailment_kw
   - site_uuid (linked to SiteSQL)

2. API endpoints for curtailment management:
   - GET /site/{site_uuid} - Get curtailments for a site
   - POST / - Create new curtailment
   - PUT /{curtailment_uuid} - Update curtailment
   - DELETE /{curtailment_uuid} - Delete curtailment

3. Forecast curtailment application:
   - Added `apply_curtailments` parameter to forecast retrieval
   - Proper handling of multiple curtailments
   - Edge case handling (midnight curtailments, etc.)

4. Comprehensive test coverage:
   - Core functionality tests
   - API endpoint tests
   - Edge case tests

Fixes #212 

## How Has This Been Tested?

The changes have been verified through:

1. Unit tests:
   - `tests/read/test_forecast_curtailment.py`
   - `tests/api/test_curtailment.py`

2. Test cases include:
   - Basic curtailment application
   - Multiple curtailments handling
   - Edge cases (midnight curtailments)
   - API endpoint validation
   - Error handling

3. Manual verification of:
   - Database migrations
   - API endpoint functionality
   - Forecast adjustment logic

### CODE REVIEW

File: pvsite_datamodel/api/curtailment.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+"""FastAPI routes for curtailment operations."""
```
Comment (peterdudfield) [inline pvsite_datamodel/api/curtailment.py:1]: Sorry, please remove this API, we dont need it

Comment (peterdudfield) [reply pvsite_datamodel/api/curtailment.py:1]: It will be going into `india-api`

File: pvsite_datamodel/read/latest_forecast_values.py [hunk 2]
```diff
@@ -22,6 +23,7 @@ def get_latest_forecast_values_by_site(
     day_ahead_hours: Optional[int] = None,
     day_ahead_timezone_delta_hours: Optional[float] = 0,
     model_name: Optional[str] = None,
+    apply_curtailments: bool = True,
```
Comment (peterdudfield) [inline pvsite_datamodel/read/latest_forecast_values.py:26]: please remove this, and below in this file. 

We are likely to apply curtailments before the forecast is saved

File: pvsite_datamodel/write/curtailment.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+"""Write operations for curtailments."""
```
Comment (peterdudfield) [inline pvsite_datamodel/write/curtailment.py:1]: this is good

File: pvsite_datamodel/sqlmodels.py [hunk 4]
```diff
@@ -473,3 +476,66 @@ class APIRequestSQL(Base, CreatedMixin):
...
+    )
+    from_time_utc = sa.Column(
+        sa.Time,
+        nullable=False,
+        comment="The start time (UTC) of daily curtailment",
+    )
+    to_time_utc = sa.Column(
+        sa.Time,
+        nullable=False,
+        comment="The end time (UTC) of daily curtailment",
```
Comment (peterdudfield) [inline pvsite_datamodel/sqlmodels.py:526]: nice

File: tests/read/test_forecast_curtailment.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+"""Tests for forecast curtailment integration."""
```
Comment (peterdudfield) [inline tests/read/test_forecast_curtailment.py:1]: sorry, please remove this file

File: alembic.ini [hunk 6]
```diff
@@ -1,75 +1,9 @@
-# A generic, single database configuration.
-
 [alembic]
```
Comment (peterdudfield) [inline alembic.ini:1]: could you undo the changes here,

File: alembic/env.py [hunk 7]
```diff
@@ -28,7 +28,10 @@
 # my_important_option = config.get_main_option("my_important_option")
 # ... etc.
 
-config.set_main_option("sqlalchemy.url", os.environ["DB_URL"])
```
Comment (peterdudfield) [inline alembic/env.py:31]: could you undo the changes here

**Comment by Dakshbir** [comment]
@peterdudfield sorry for the delay sir. I was occupied with some college related stuff. 
I am fixing the issues which you have reviewed and told.

**Comment by Dakshbir** [comment]
I have done the changes. Now running the checks

**Comment by Dakshbir** [comment]
@peterdudfield hello sir
I have done the changes which you asked for in the files but I am still facing some issues while passing the checks. I am unable to pass the checks. 
could you help me with this?

**Comment by Dakshbir** [comment]
@peterdudfield hello sir
I have done the changes which you asked for in the files but I am still facing some issues while passing the checks. I am unable to pass the checks. 
could you help me with this?
